include/abb_librws/rws_cfg.h

```diff
@@ -0,0 +1,263 @@
+ *      and the following disclaimer in the documentation
+ *      and/or other materials provided with the
+ *      distribution.
+ *    * Neither the name of ABB nor the names of its
+ *      contributors may be used to endorse or promote
+ *      products derived from this software without
+ *      specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+ * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ ***********************************************************************************************************************
+ */
+
+#ifndef RWS_CFG_H
+#define RWS_CFG_H
+
+#include <string>
+
+#include "rws_rapid.h"
+
+namespace abb
+{
+namespace rws
+{
+/**
+ * \brief Namespace for configuration domains.
+ */
+namespace cfg
+{
+
+/**
+ * \brief Namespace for types belonging to the motion (MOC) domain.
+ */
+namespace moc
+{
+/**
+ * \brief Representation of a 'Arm' instance.
+ */
+struct Arm
+{
+  /**
+   * \brief The instance's name.
+   */
+  std::string name;
+
+  /**
+   * \brief The lower joint bound.
+   *
+   * The unit depends on the type of joint, i.e. [rad] for rotational and [m] for linear.
+   *
+   */
+  double lower_joint_bound;
+
+  /**
+   * \brief The upper joint bound.
+   *
+   * The unit depends on the type of joint, i.e. [rad] for rotational and [m] for linear.
+   */
+  double upper_joint_bound;
+};
+
+/**
+ * \brief Representation of a 'Joint' instance.
+ */
+struct Joint
+{
+  /**
+   * \brief The instance's name.
+   */
+  std::string name;
+
+  /**
+   * \brief The joint's logical axis.
+   */
+  int logical_axis;
+
+  /**
+   * \brief The joint's kinematic axis number.
+   */
+  int kinematic_axis_number;
+
+  /**
+   * \brief The arm used by the joint.
+   */
+  std::string use_arm;
+
+  /**
+   * \brief The transmission used by the joint.
+   */
+  std::string use_transmission;
+};
+
+/**
+ * \brief Representation of a 'Mechanical Unit' instance.
+ */
+struct MechanicalUnit
+{
+  /**
+   * \brief The instance's name.
+   */
+  std::string name;
+
+  /**
+   * \brief The robot used by the mechanical unit.
+   */
+  std::string use_robot;
+
+  /**
+   * \brief The singles used by the mechanical unit.
+   */
+  std::vector<std::string> use_singles;
```

Comment:
What do we store in this `vector`? Names associated with objects?

Could we use instances of actual types here instead?

Reply:
Yes, it stores the names, and again, this is how the `RobotWare` configurations are structured.

I think it is more reasonable to leave the "parsing" up to actual use cases. This is manily to structure data on the same format as `RobotWare` does.